detector/src/detector.cpp

```diff
@@ -3,82 +3,87 @@
 
 namespace ariitk::detector {
 
-Detector::~Detector() {
-    cv::destroyAllWindows();
-}
+Detector::~Detector() { cv::destroyAllWindows(); }
 
 void Detector::setCannyParams(const int& l, const int& u, const int& s) {
-    canny_param_low_ = l;
-    canny_param_upper_ = u;
-    canny_kernel_size_ = s;
+	canny_param_low_ = l;
+	canny_param_upper_ = u;
+	canny_kernel_size_ = s;
 }
 
 void Detector::thresholdImage(cv::Mat& img) {
-    if(img.empty()) { return; }
-    
-    cv::Mat blur_img(img.size(), CV_8UC3);
-    cv::Mat hsv_img(img.size(), CV_8UC3);
-
-    cv::GaussianBlur(img, blur_img, cv::Size(3,3), 0, 0);
-    cv::cvtColor(blur_img, hsv_img, CV_BGR2HSV);
-    cv::inRange(hsv_img, hsv_min_, hsv_max_, thresh_img_);
+	if (img.empty()) { return; }
+
+	cv::Mat blur_img(img.size(), CV_8UC3);
+	cv::Mat hsv_img(img.size(), CV_8UC3);
+
+	cv::GaussianBlur(img, blur_img, cv::Size(3, 3), 0, 0);
+	cv::cvtColor(blur_img, hsv_img, CV_BGR2HSV);
+	cv::inRange(hsv_img, hsv_min_, hsv_max_, thresh_img_);
 }
 
 void Detector::findGoodContours() {
-    cv::Mat canny_img(thresh_img_.size(), CV_8UC1);
-    
-    cv::Canny(thresh_img_, canny_img, canny_param_low_, canny_param_upper_, canny_kernel_size_);
+	cv::Mat canny_img(thresh_img_.size(), CV_8UC1);
 
-    std::vector<std::vector<cv::Point>> contours;
-    std::vector<cv::Vec4i> hierarchy;
-    cv::findContours(thresh_img_, contours, hierarchy, CV_RETR_TREE, CV_CHAIN_APPROX_SIMPLE);
+	cv::Canny(thresh_img_, canny_img, canny_param_low_, canny_param_upper_, canny_kernel_size_);
 
-    good_contours_.clear();
-    for(auto& contour : contours) { if(cv::contourArea(contour) > min_contour_area_) { good_contours_.push_back(contour); } }
-}
+	good_contours_.clear();
+	good_contours_.shrink_to_fit();
```

Comment:
clearing `good_contours` twice here. remove these two lines

Reply:
clear() only removes all the elements from the vector but it still has some capacity. shrink_to_fit() deallocates the memory too

Reply:
agreed, but my point was that there's another clear right after it. so that's redundant, you can remove either one of the two